torch/_dynamo/variables/torch.py

```diff
@@ -608,11 +608,27 @@ def fn_with_prim_types(x):
                 and self.value in tensortype_to_dtype
                 and len(args) == 1
                 and isinstance(args[0], ListVariable)
-                and args[0].is_python_constant()
             ):
+                all_ndarray = True
                 for x in args[0].items:
-                    if isinstance(x.value, np.generic):
+                    if not isinstance(x, variables.NumpyNdarrayVariable):
+                        all_ndarray = False
+                    if x.is_python_constant() and isinstance(
+                        x.as_python_constant(), np.generic
+                    ):
                         x.value = x.value.item()
+                if all_ndarray and kwargs == {}:
+                    # Stack FakeTensor
+                    stacked = wrap_fx_proxy(
+                        tx=tx,
+                        proxy=tx.output.create_proxy(
+                            "call_function",
+                            torch.stack,
+                            *proxy_args_kwargs(args, kwargs),
+                        ),
+                        **options,
+                    )
+                    args[0] = stacked
```

Comment:
This can be seen from outside this function so it's not great. Better create a new `args`.

Reply:
Technically, we already mutate the args in main `x.value = x.value.item()`. 

This is actually worse. 

I think it's better if we just torch.stack everything, even single-elem np.generic Tensors.

I think we can get away with shallow copying the list and then mutating it freely.